docs/persistent_compilation_cache.md

```diff
@@ -66,110 +94,81 @@ Cloud Storage (GCS) bucket. We recommend the following configuration:
 
 *  All encryption policies are supported.
 
-Assuming that `gs://jax-cache` is the GCS bucket, set `cache-location` as
+Assuming that `gs://jax-cache` is the GCS bucket, set cache location as
 follows:
 
 ```python
-import jax
-
 jax.config.update("jax_compilation_cache_dir", "gs://jax-cache")
 ```
 
 ## How it works
 
-The JAX compilation cache works by hashing a number of parameters to create a signature for a compiled function. These are:
+The cache key is the signature for a compiled function containing the
+following parameters:
 
 * The computation performed by the function captured by the non-optimized HLO of the JAX function being hashed
 
-* The jaxlib version 
+* The jaxlib version
 
 * Relevant XLA compilation flags
 
-* Device configuration captured in general, by the number of devices and the topology of the devices. 
+* Device configuration captured in general, by the number of devices and the topology of the devices.
   Currently for GPUs, the topology only contains a string representation of the GPU name
 
 * Compression algorithm used to compress the compiled executable
 
 * Any custom hooks
```

Comment:
This is probably unclear to users. Do you think it might be worth explaining what hooks are in this context?

Reply:
It's defined in https://github.com/google/jax/blob/c3bc88d5e4e3b25d99b7aa9ec2b1c93a854d976c/jax/_src/cache_key.py#L55 and the tests are in https://github.com/google/jax/blob/0b28a4b168115f02b7074a7702505f11c69747e5/tests/cache_key_test.py#L130-L143. I will add some explanations.

Reply:
Done